### ISSUE TITLE
Adding support for channels

### ISSUE THREAD

**Issue by cdvv7788** [pydanny/cookiecutter-django#1058]
We should have the option to ask users if they want to use channels, just like we do with celery.
Setting up channels should be easy, using the in memory layer locally and redis on production (or redis on both). Not much has to be assumed and is easy to implement.

Thoughts?

**Comment by burhan** [comment]
My only concern is that its not really fully tested on Windows (a platform we support); I'd have to run it through some tests on Windows before I would be comfortable with this.

**Comment by cdvv7788** [comment]
I have not worked on windows for years, so i wouldn't know about channels support. @burhan how can i help to test that?

**Comment by pydanny** [comment]
Channels also doesn't work on Elastic Beanstalk. As incredibly awesome as that platform is, it still forces you to use just `mod_wsgi`. 😢 

As for windows support, if there's issues, we might be able to use this as a way to connect with the Django Channels and get better support for it in Windows. For what it's worth, I know @andrewgodwin uses Windows for programming that isn't Django/Python.

**Comment by burhan** [comment]
Last time I spoke with @andrewgodwin he mentioned he mostly uses the embedded linux runtime that comes with Windows (I wanted to get some pointers for Windows development as I too have to deal with it for work).

**Comment by andrewgodwin** [comment]
Yes, I do all my development on Windows inside the Linux Subsystem for Windows, which is basically just another UNIX with Ubuntu userland. It's entirely different from running under Win32 windows because it's just Linux-compatible, so things just work.

**Comment by cdvv7788** [comment]
I just ran a project with channels on windows with linux subsystem, just like @andrewgodwin. Works as expected. Do we need someone to test this directly on windows before proceeding?

**Comment by pydanny** [comment]
Not every Windows user of Cookiecutter Django runs the Linux Subsystem for Windows. If we expect all our Windows users to utilize this functionality, your pull request will need:

* Documentation that references the official guides for the Linux Subsystem for Windows
* Documentation on how to use Cookiecutter Django in the Linux Subsystem for Windows
* A logic branch during the rendering process that doesn't allow Channels to be used with Elastic Beanstalk setups

Personally, I would love to see this PR happen. We might even be able to get some oversight on the pull request from the MS Azure or VS Studio team.

**Comment by cdvv7788** [comment]
Ok, i get your point @pydanny. I will test this directly on windows before proceeding, otherwise I will start working following those guidelines.

**Comment by philippeluickx** [comment]
Beanstalk now supports multicontainer deployments. Therefore, we could probably use Channels through Docker.
I am messing around at the moment to setup django-cookiecutter through multicontainer Docker on Beanstalk, but all development is on hold now due to a (silly) bug - https://forums.aws.amazon.com/thread.jspa?threadID=251051

**Comment by pydanny** [comment]
@philippeluickx, according to https://forums.aws.amazon.com/thread.jspa?threadID=251051 it looks like the bug is fixed. If you could get ELB to work with multicontainer Docker we would make a PR for this project an absolute priority.

**Comment by philippeluickx** [comment]
I managed to get cookiecutter run on multicontainer docker in EBS: updated ticket https://github.com/pydanny/cookiecutter-django/issues/1004

**Comment by kinwolfqc** [comment]
Hi, trying to run channels 2 with cookiecutter and docker but I am running in a weird problem. What I did is start a regular project with cookicutter than used channels 2 instructions to integrate it in django(https://channels.readthedocs.io/en/latest/installation.html)  Then I also edited the docker-compose file so it runs "runserver" instead of "runserver-plus"

But channels is still complaining with: 

> django.core.exceptions.ImproperlyConfigured: Cannot find 'application' in ASGI_APPLICATION module second_season_channels.routing

What is weird is that it obviously found the routing.py file, but cannot see "application" within it, but it's there.  The content is taken directly from the doc

`from channels.routing import ProtocolTypeRouter`
`application = ProtocolTypeRouter({`
   ` # Empty for now (http->django views is added by default)`
`})`

Is there something else that must be modified in the django project basic config to make it work correctly?

**Comment by yusufhilmi** [comment]
Is there any improvements or personal projects on this subject. I am a newbie to django and I will try to go for production soon. I am using channels in my project.  I would be glad if anyone could show me the best possible solution. 
By the way, thanks for all the work that has been done until now, experiencing with open-source and contributions from all over the world always amaze me:)
